lib/srv/desktop/rdp/rdpclient/client.go

```diff
@@ -167,14 +167,19 @@ func New(ctx context.Context, cfg Config) (*Client, error) {
 // Run starts the rdp client and blocks until the client disconnects,
 // then ensures the cleanup is run.
 func (c *Client) Run(ctx context.Context) error {
-	defer c.close()
```

Comment:
Should we keep the previous behavior and do some cleaning (done on the `close` function) when [`connect`](https://github.com/gravitational/teleport/pull/14323/files#diff-27da86cebdfa8870a5999b5cb051103a7cfc23e41131139feab9b26e935a5915R170) fails?

Reply:
Need to cleanup the c.handle regardless of whether the connection succeeds. Good catch!

Reply:
Yes good call, but also we can't call `C.free_rdp(c.rustClient)` if `connect` fails. See https://github.com/gravitational/teleport/pull/14323/commits/8a1db5119bded529581054ecf3ad684ff8ec5b58 where I centralize both allocations and add some checks in `cleanup()`.